### PR TITLE
ss-reports-get CLI

### PR DESCRIPTION
**Pull request by konstan**
Download reports of a deployment.

Connected to #272

### CODE REVIEW

File: client/src/main/python/slipstream/resources/reports.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+from slipstream.HttpClient import HttpClient
+import slipstream.util as util
+from slipstream.NodeDecorator import NodeDecorator
+
+
+def _download(username, password, src_url, dst_file):
+    request = urllib2.Request(src_url)
+    request.add_header('Authorization',
+                       (b'Basic ' + (username + b':' + password).encode('base64')).replace('\n', ''))
```
Comment (schaubl) [inline client/src/main/python/slipstream/resources/reports.py]: I think `Basic` authentication has been deprecated in SlipStream.
@st @loomis: do you confirm ?

Comment (st) [inline client/src/main/python/slipstream/resources/reports.py]: Deprecation means it'll be removed in the near future.

Comment (schaubl) [inline client/src/main/python/slipstream/resources/reports.py]: So I think we should use cookie auth for new stuff (and this is a new command 😉)

Comment (st) [inline client/src/main/python/slipstream/resources/reports.py]: `basic` authentication is actually marked as `deprecated` to indicate that in a near future, it'll removed.
Resources under `/api` namespace already do *not* support `basic` auth.

Comment (schaubl) [inline client/src/main/python/slipstream/resources/reports.py]: So I think we should use cookie auth for new stuff (and this is a new command 😉)

Comment (konstan) [inline client/src/main/python/slipstream/resources/reports.py]: @schaubl Please check 27eccef

**Comment by konstan** [comment]
The function was made generic and moved to utils.  Yes, the basic auth still there but is used as a fallback solution, with cookie as the first option.  Cookie is assumed to be provided by HttpClient and again is the first choice for authn when downloading.  So, by doing this I just offloaded the issue to HttpClient as this is not the main issue of this ticket.  Please comment.

**Comment by konstan** [comment]
@schaubl As agreed added the issue for migration to cookie based authn in the client. 
https://github.com/slipstream/SlipStreamClient/issues/279

**Pull request merged**